presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/SelectionContext.java

```diff
@@ -22,14 +23,16 @@
     private final boolean authenticated;
     private final String user;
     private final Optional<String> source;
+    private final Set<String> clientTags;
     private final int queryPriority;
     private final Optional<String> queryType;
 
-    public SelectionContext(boolean authenticated, String user, Optional<String> source, int queryPriority, Optional<String> queryType)
+    public SelectionContext(boolean authenticated, String user, Optional<String> source, Set<String> clientTags, int queryPriority, Optional<String> queryType)
     {
         this.authenticated = authenticated;
         this.user = requireNonNull(user, "user is null");
         this.source = requireNonNull(source, "source is null");
+        this.clientTags = requireNonNull(clientTags, "tags is null");
```

Comment:
`this.clientTags = Collections.unmodifiableSet(requireNonNull(clientTags, "tags is null"));`